File: esport_fantasy_back/src/main/java/com/epicode/Spring/model/Match.java
package com.epicode.Spring.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table (name = "Matches")
public class Match {

	@Id
	@GeneratedValue (strategy = GenerationType.IDENTITY)
	private Long id;
	
	@ManyToOne
    @JoinColumn(name = "team_b_id")
    private Team teamBId;
	
	@ManyToOne
    @JoinColumn(name = "team_r_id")
    private Team teamRId;
	
	private Integer dayOfGames;
	
}

File: esport_fantasy/src/app/interface/Match.ts
export interface Match {
  id: number;
  dayOfGames: number;
  teamRId:  {
    id: number,
    league: String,
    name: String,
    nationality: String,
  };
  teamBId: {
    id: number,
    league: String,
    name: String,
    nationality: String,
  };
}
